.gitignore

```diff
@@ -3,4 +3,40 @@ cmake-build-debug/
 libat.*
 .idea/*
 *.xml
-/distributions/JavaScript/output/
\ No newline at end of file
+/distributions/JavaScript/output/
+
+#object files
+*.o 
+
+#libraries
+*.lo
+*.la 
+
+*.deps/
+*.libs/
+
+*Makefile
+*Makefile.in
```

Comment:
Check if all `*in` files should stay

Reply:
these files are autogenerated by makefile.am